src/ImageSharp.Web/Caching/PhysicalFileSystemCache.cs

```diff
@@ -146,5 +196,40 @@ public async Task<DateTimeOffset> SetAsync(string key, Stream stream)
         /// <param name="key">The cache key.</param>
         /// <returns>The <see cref="string"/>.</returns>
         private string ToFilePath(string key) => $"{this.Settings[Folder]}/{string.Join("/", key.Substring(0, (int)this.options.CachedNameLength).ToCharArray())}/{key}";
+
+        private async Task<Dictionary<string, string>> ReadMetadataFile(Stream stream)
```

Comment:
This would have to be implementation agnostic. 

I'm thinking we create a custom struct that can be easily serialized deserialized into raw bytes that can be used for storing both the content type and last modified date. 

CacheMeta
8 bytes | ModifiedDate
2 bytes | Content Type string length
n bytes | Content Type

Reply:
My thinking with this implementation was to use a file format that provides easy forward/backward compatibility so that additional metadata fields could be easily added in the future (I already have one such use that I would like to discuss if you end up deciding to proceed with this metadata concept).  I would have used JSON, but didn't want to take a dependency on a JSON library, so I went with a really simple text-based key/value pair file format.  I wasn't too worried about wasting a few extra bytes by having the data stored as text as the size should be trivial relative to the actual image data that it is annotating.

Reply:
But thinking about it now, I do like the idea of having the metadata contained in a struct, as that would make it possible to add new metadata fields without needing to add any additional methods or change function signatures.  Something like:

```
public struct ImageMetadata
{
    public DateTime LastModifiedUtc;
    public string ContentType;
}

public interface IImageResolver
{
    Task<Stream> OpenReadAsync();
    Task<ImageMetadata> GetImageMetadata();
}

public interface IImageCache
{
    Task<IImageResolver> GetAsync(string key);
    Task<DateTimeOffset> SetAsync(string key, Stream stream, ImageMetadata metadata);
}
```